turbopack/crates/turbopack-ecmascript/src/analyzer/imports.rs

```diff
@@ -709,10 +705,15 @@ impl Visit for Analyzer<'_> {
         m.visit_children_with(self);
     }
 
-    fn visit_stmt(&mut self, n: &Stmt) {
-        if self.comments.is_some() {
-            // only visit children if we potentially need to mark import / requires
-            n.visit_children_with(self);
+    fn visit_function(&mut self, n: &Function) {
+        let old_has_bound_this = self.has_bound_this;
+        self.has_bound_this = true;
```

Comment:
you need to do a similar trick for class members and object methods

```
{foo() { return this;}}
class {
  foo() { return this;
}
```

unless those automatically call through to visit_function?

Reply:
Nice catch!.

`visit_function` is not called for other AST types than those storing `Function`. It's sort of limit of codegen-ed visitor.